src/test/java/seedu/address/logic/commands/FindFreeTimeCommandTest.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalEvents.getTypicalSochedule;
+
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.common.Date;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteTaskCommand}.
+ */
+public class FindFreeTimeCommandTest {
+
+    private Model model = new ModelManager(getTypicalSochedule(), new UserPrefs());
+
+    @Test
+    public void execute_validDateNoFreeTime_success() {
+        Date exampleDate = new Date("2021-03-16");
+        FindFreeTimeCommand findFreeTimeCommand = new FindFreeTimeCommand(exampleDate);
+        String expectedMessage = findFreeTimeCommand.MESSAGE_NO_FREE_TIME;
+        try {
+            CommandResult result = findFreeTimeCommand.execute(model);
+            assertEquals(result, new CommandResult(expectedMessage));
+        } catch (CommandException ce) {
+            throw new AssertionError("Execution of command should not fail.", ce);
+        }
+    }
+
+    @Test
+    public void execute_validDateAllDayFree_success() {
+        Date exampleDate = new Date("2025-03-16");
+        ArrayList<String> actual = model.getFreeTimeSlots(exampleDate);
+        ArrayList<String> expected = new ArrayList<>();
+        expected.add("The entire day is free!");
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void equals() {
+        Date exampleDateOne = new Date("2025-03-16");
+        Date exampleDateTwo = new Date("2025-03-17");
```

Comment:
Once these magic dates are put in CommandTestUtil, you can just import and use them without initialising them again.